### PR TITLE
Uploading folder and csv file in training and validation phases

### LINKED ISSUE #496: Select image data and provide labels as .csv
## Feature

In the web client, training a model with image data requires the user to select the images every time for every label, which can quickly get cumbersome. A user should be able to provide all the images at once, along with a single .csv file containing all the image file names mapped to their corresponding label.

## Specifications
A user would provide a single .csv file with lines of the following format: `image_file_name.png, label`

### PR DESCRIPTION
**Pull request by francesco98**
### Discojs library
Added dependency to d3 package

### Web-Client
Updating DatasetInput, IconCard, FileSelection components for allowing CSV and folder upload.
You can still upload by label (as before), or you can choose to upload a folder + CSV file.

Screenshot 2022-10-30 at 17 24 11
Screenshot 2022-10-30 at 17 24 03

### CODE REVIEW

**Comment by francesco98** [comment]
I have issues with WRTC on MAC M1, so I cannot correctly update the `package-lock.json`. @s314cy could you please pull the branch and update the file accordingly? Thank you!

**Comment by francesco98** [comment]
No idea why `test-lib-core` is failing. @s314cy Do you have any idea?

**Review by martinjaggi — APPROVED** [review]

File: web-client/src/components/data/dataset_input/FileSelection.vue [hunk 1]
```diff
@@ -100,29 +122,44 @@
...
+
+const props = withDefaults(defineProps<Props>(), {
+  isDirectory: false,
+  isMultiple: true,
+  acceptFiles: () => ['*'],
+  infoText: ''
+})
 
-const nbrSelectedFiles = ref(0)
+console.log(props.isDirectory)
```
Comment (s314cy) [inline web-client/src/components/data/dataset_input/FileSelection.vue]: here too :p

File: web-client/src/components/data/dataset_input/DatasetInput.vue [hunk 2]
```diff
@@ -63,9 +144,48 @@ interface Props {
...
+        if (imageFile) {
+          props.datasetBuilder.addFiles([imageFile], row.label)
+        }
+      })
+    }
+  } else {
+    props.datasetBuilder.addFiles(filesArray)
+  }
+
+  console.log(props.datasetBuilder.size())
```
Comment (s314cy) [inline web-client/src/components/data/dataset_input/DatasetInput.vue]: don't forget to remove the dev console logs :)

**Review by s314cy — CHANGES_REQUESTED** [review]
excellent work, the UI/UX is on point! 🚀 

regarding the ability of discojs to read labels from a separate .csv file, it would be nice to have the functionality available within the lib directly, not only in the web client

this is why, in my opinion, the code handling the csv read and passing the labels to the dataset builder should be included in the data loader/dataset builder code _directly_

I can help you integrate this to the adequate classes, if you need any help!

File: discojs/package.json [hunk 3]
```diff
@@ -17,8 +17,10 @@
     "@koush/wrtc": "0.5",
     "@tensorflow/tfjs": "4",
     "@tensorflow/tfjs-node": "4",
+    "@types/d3": "^7.4.0",
     "@types/msgpack-lite": "0.1",
     "axios": "0.27",
+    "d3": "^7.6.1",
```
Comment (s314cy) [inline discojs/package.json]: following my code review, it's actually relevant that you added d3 to `discojs`'s dependencies, although your current code would only require the dependency within `web-client`

**Pull request closed without merging**